packages/sync/src/modules/class-full-sync-immediately.php

```diff
@@ -0,0 +1,405 @@
+		return $this->set_status( wp_parse_args( $values, $this->get_status() ) );
+	}
+
+	/**
+	 * Retrieve the status of the current full sync.
+	 *
+	 * @param array $values New values to set.
+	 *
+	 * @access public
+	 *
+	 * @return boolean Full sync status.
+	 */
+	public function set_status( $values ) {
+		return \Jetpack_Options::update_raw_option( self::STATUS_OPTION, $values );
+	}
+
+	/**
+	 * Given an initial Full Sync configuration set the initial status.
+	 *
+	 * @param array $full_sync_config Full sync configuration.
+	 *
+	 * @return array Initial Sent status.
+	 */
+	public function get_initial_progress( $full_sync_config ) {
+		// Set default configuration, calculate totals, and save configuration if totals > 0.
+		$status = array();
+		foreach ( $full_sync_config as $name => $config ) {
+			$module          = Modules::get_module( $name );
+			$status[ $name ] = array(
+				'total'    => $module->total( $config ),   // Total.
+				'sent'     => 0,             // Sent.
+				'finished' => false,
+			);
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Get the range for content (posts and comments) to sync.
+	 *
+	 * @access private
+	 *
+	 * @return array Array of range (min ID, max ID, total items) for all content types.
+	 */
+	private function get_content_range() {
+		$range  = array();
+		$config = $this->get_status()['config'];
+		// Only when we are sending the whole range do we want to send also the range.
+		if ( true === isset( $config['posts'] ) && $config['posts'] ) {
+			$range['posts'] = $this->get_range( 'posts' );
+		}
+
+		if ( true === isset( $config['comments'] ) && $config['comments'] ) {
+			$range['comments'] = $this->get_range( 'comments' );
+		}
+
+		return $range;
+	}
+
+	/**
+	 * Get the range (min ID, max ID and total items) of items to sync.
+	 *
+	 * @access public
+	 *
+	 * @param string $type Type of sync item to get the range for.
+	 *
+	 * @return array Array of min ID, max ID and total items in the range.
+	 */
+	public function get_range( $type ) {
+		global $wpdb;
+		if ( ! in_array( $type, array( 'comments', 'posts' ), true ) ) {
+			return array();
+		}
+
+		switch ( $type ) {
+			case 'posts':
+				$table     = $wpdb->posts;
+				$id        = 'ID';
+				$where_sql = Settings::get_blacklisted_post_types_sql();
+
+				break;
+			case 'comments':
+				$table     = $wpdb->comments;
+				$id        = 'comment_ID';
+				$where_sql = Settings::get_comments_filter_sql();
+				break;
+		}
+
+		// TODO: Call $wpdb->prepare on the following query.
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$results = $wpdb->get_results( "SELECT MAX({$id}) as max, MIN({$id}) as min, COUNT({$id}) as count FROM {$table} WHERE {$where_sql}" );
+		if ( isset( $results[0] ) ) {
+			return $results[0];
+		}
+
+		return array();
+	}
+
+	/**
+	 * Continue sending.
+	 *
+	 * @access public
+	 */
+	public function continue_enqueuing() {
+		$this->continue_sending();
+	} // Seconds.
```

Comment:
Not sure what's going on here.

Reply:
we are still using the old logic that controls enqueuing to control sending https://github.com/Automattic/jetpack/blob/1b981c3573c82a4c83b3c212a241b59097691792/packages/sync/src/class-sender.php#L280 we want to touch as little files as possible so it was easier to do this hack than to add `if`s and calls to `continue_sending`.  Once we deprecate queue full sync we will need to remove this method and call `continue_sending` directly.

Reply:
Gotcha. Maybe the `//seconds` comment could be removed though, it feels odd.

Reply:
```suggestion
	}
```

Looks like it could be coming from a bad rebase? 🤷‍♂